diff --git a/src/core/hle/service/olsc/remote_storage_controller.cpp b/src/core/hle/service/olsc/remote_storage_controller.cpp
index 7e04b7c042..9c8057add6 100644
--- a/src/core/hle/service/olsc/remote_storage_controller.cpp
+++ b/src/core/hle/service/olsc/remote_storage_controller.cpp
@@ -28,7 +28,7 @@ IRemoteStorageController::IRemoteStorageController(Core::System& system_)
         {15, nullptr, "RegisterUploadSaveDataTransferTaskForAutonomyRegistration"},
         {16, nullptr, "CreateCleanupToDeleteSaveDataArchiveInfoTask"},
         {17, nullptr, "ListDataInfo"},
-        {18, nullptr, "GetDataInfo"},
+        {18, D<&IRemoteStorageController::GetDataInfo>, "GetDataInfoV1"},
         {19, nullptr, "GetDataInfoCacheUpdateNativeHandleHolder"},
         {20, nullptr, "CreateSaveDataArchiveInfoCacheForSaveDataBackupUpdationTask"},
         {21, nullptr, "ListSecondarySaves"},
@@ -37,7 +37,7 @@ IRemoteStorageController::IRemoteStorageController(Core::System& system_)
         {24, nullptr, "GetSecondarySaveDataInfo"},
         {25, nullptr, "RegisterDownloadSaveDataTransferTaskForAutonomyRegistration"},
         {26, nullptr, "Unknown26"}, //20.0.0+
-        {27, nullptr, "Unknown27"}, //20.0.0+
+        {27, D<&IRemoteStorageController::GetDataInfo>, "GetDataInfoV2"}, //20.0.0+
         {28, nullptr, "Unknown28"}, //20.0.0+
         {29, nullptr, "Unknown29"}, //21.0.0+
         {800, nullptr, "Unknown800"}, //20.0.0+
@@ -60,4 +60,10 @@ Result IRemoteStorageController::GetSecondarySave(Out<bool> out_has_secondary_sa
     R_SUCCEED();
 }
 
+Result IRemoteStorageController::GetDataInfo(Out<std::array<u8, 0x38>> out_data, u64 application_id) {
+    LOG_WARNING(Service_OLSC, "(STUBBED) called, application_id={:016X}", application_id);
+    out_data->fill(0);
+    R_SUCCEED();
+}
+
 } // namespace Service::OLSC
diff --git a/src/core/hle/service/olsc/remote_storage_controller.h b/src/core/hle/service/olsc/remote_storage_controller.h
index e7a0b52442..ac402e0660 100644
--- a/src/core/hle/service/olsc/remote_storage_controller.h
+++ b/src/core/hle/service/olsc/remote_storage_controller.h
@@ -14,6 +14,7 @@ public:
 private:
     Result GetSecondarySave(Out<bool> out_has_secondary_save, Out<std::array<u64, 3>> out_unknown,
                             u64 application_id);
+    Result GetDataInfo(Out<std::array<u8, 0x38>> out_data, u64 application_id);
 };
 
 } // namespace Service::OLSC
