diff --git a/src/common/settings.cpp b/src/common/settings.cpp
index b7c2d3c9a9..f6990107ba 100644
--- a/src/common/settings.cpp
+++ b/src/common/settings.cpp
@@ -348,7 +348,7 @@ void TranslateResolutionInfo(ResolutionSetup setup, ResolutionScalingInfo& info)
     }
     info.up_factor = static_cast<f32>(info.up_scale) / (1U << info.down_shift);
     info.down_factor = static_cast<f32>(1U << info.down_shift) / info.up_scale;
-    info.active = info.up_scale != 1 || info.down_shift != 0;
+    info.active = true;
 }
 
 void UpdateRescalingInfo() {
