diff --git a/src/common/settings_enums.h b/src/common/settings_enums.h
index d4935d9b6d..0f78efaf51 100644
--- a/src/common/settings_enums.h
+++ b/src/common/settings_enums.h
@@ -129,7 +129,7 @@ ENUM(TimeZone, Auto, Default, Cet, Cst6Cdt, Cuba, Eet, Egypt, Eire, Est, Est5Edt
 ENUM(AnisotropyMode, Automatic, Default, X2, X4, X8, X16, X32, X64, None);
 ENUM(AstcDecodeMode, Cpu, Gpu, CpuAsynchronous);
 ENUM(AstcRecompression, Uncompressed, Bc1, Bc3);
-ENUM(FramePacingMode, Target_Auto, Target_30, Target_60, Target_90, Target_120);
+ENUM(FramePacingMode, Target_Auto, Target_20, Target_30, Target_40, Target_50, Target_60, Target_90, Target_120);
 ENUM(VSyncMode, Immediate, Mailbox, Fifo, FifoRelaxed);
 ENUM(VramUsageMode, Conservative, Aggressive);
 ENUM(RendererBackend, OpenGL_GLSL, Vulkan, Null, OpenGL_GLASM, OpenGL_SPIRV);
diff --git a/src/video_core/renderer_vulkan/vk_scheduler.h b/src/video_core/renderer_vulkan/vk_scheduler.h
index 2913211480..a25fb3bcd9 100644
--- a/src/video_core/renderer_vulkan/vk_scheduler.h
+++ b/src/video_core/renderer_vulkan/vk_scheduler.h
@@ -116,19 +116,18 @@ public:
     void Wait(u64 tick, double target_fps = 0.0) {
         if (Settings::values.use_speed_limit.GetValue() && target_fps > 0.0) {
             const auto now = std::chrono::steady_clock::now();
-            if (start_time == std::chrono::steady_clock::time_point{} || current_target_fps != target_fps) {
-                start_time = now;
-                frame_counter = 0;
+            if (next_frame_time == std::chrono::steady_clock::time_point{} || current_target_fps != target_fps) {
+                next_frame_time = now;
                 current_target_fps = target_fps;
+                frame_interval = std::chrono::duration_cast<std::chrono::steady_clock::duration>(std::chrono::duration<double, std::micro>(1'000'000.0 / current_target_fps));
             }
-            frame_counter++;
-            std::chrono::duration<double> frame_interval(1.0 / current_target_fps);
-            auto target_time = start_time + frame_interval * frame_counter;
-            if (target_time > now) {
-                std::this_thread::sleep_until(target_time);
+            next_frame_time += frame_interval;
+            if (next_frame_time > now) {
+                std::this_thread::sleep_until(next_frame_time);
             } else {
-                start_time = now;
-                frame_counter = 0;
+                while (next_frame_time <= now) {
+                    next_frame_time += frame_interval;
+                }
             }
         }
         if (tick > 0) {
@@ -281,8 +280,8 @@ private:
     std::condition_variable_any event_cv;
     std::jthread worker_thread;
 
-    std::chrono::steady_clock::time_point start_time{};
-    u64 frame_counter{};
+    std::chrono::steady_clock::time_point next_frame_time{};
+    std::chrono::steady_clock::duration frame_interval{};
     double current_target_fps{};
 };
 
diff --git a/src/video_core/renderer_vulkan/vk_swapchain.cpp b/src/video_core/renderer_vulkan/vk_swapchain.cpp
index 89aa6c4628..3e6c5dd12a 100644
--- a/src/video_core/renderer_vulkan/vk_swapchain.cpp
+++ b/src/video_core/renderer_vulkan/vk_swapchain.cpp
@@ -198,9 +198,18 @@ bool Swapchain::AcquireNextImage() {
     case Settings::FramePacingMode::Target_Auto:
         scheduler.Wait(resource_ticks[image_index]);
         break;
+    case Settings::FramePacingMode::Target_20:
+        scheduler.Wait(resource_ticks[image_index], 20.0);
+        break;
     case Settings::FramePacingMode::Target_30:
         scheduler.Wait(resource_ticks[image_index], 30.0);
         break;
+    case Settings::FramePacingMode::Target_40:
+        scheduler.Wait(resource_ticks[image_index], 40.0);
+        break;
+    case Settings::FramePacingMode::Target_50:
+        scheduler.Wait(resource_ticks[image_index], 50.0);
+        break;
     case Settings::FramePacingMode::Target_60:
         scheduler.Wait(resource_ticks[image_index], 60.0);
         break;
